content/en/docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -90,57 +119,133 @@ You can develop a KMS plugin gRPC server using a stub file available for Go. For
+  server will coast on the last valid state for about three minutes.  Thus if a user wants to take a passive approach
+  to storage migration (i.e. by waiting), they must schedule a migration to occur at `3 + N + M` minutes after the
+  remote KEK has been rotated (`N` is how long it takes the plugin to observe the `key_id` change and `M` is the
+  desired buffer to allow config changes to be processed - a minimum `M` of five minutes is recommend).  Note that no
+  API server restart is required to perform KEK rotation.
+
+  {{< caution >}}  
+  Because you don't control the number of writes performed with the DEK, we recommend rotating the KEK at least every 90 days.
+  {{< /caution >}}
 
 * protocol: UNIX domain socket (`unix`)
 
-  The plugin is implemented as a gRPC server that listens at UNIX domain socket. The plugin deployment should create a file on the file system to run the gRPC unix domain socket connection. The API server (gRPC client) is configured with the KMS provider (gRPC server) unix domain socket endpoint in order to communicate with it. An abstract Linux socket may be used by starting the endpoint with `/@`, i.e. `unix:///@foo`. Care must be taken when using this type of socket as they do not have concept of ACL (unlike traditional file based sockets). However, they are subject to Linux networking namespace, so will only be accessible to containers within the same pod unless host networking is used. 
+  The plugin is implemented as a gRPC server that listens at UNIX domain socket. 
+  The plugin deployment should create a file on the file system to run the gRPC unix domain socket connection. 
+  The API server (gRPC client) is configured with the KMS provider (gRPC server) unix 
+  domain socket endpoint in order to communicate with it. 
+  An abstract Linux socket may be used by starting the endpoint with `/@`, i.e. `unix:///@foo`. 
+  Care must be taken when using this type of socket as they do not have concept of ACL 
+  (unlike traditional file based sockets). 
+  However, they are subject to Linux networking namespace, so will only be accessible to 
+  containers within the same pod unless host networking is used.
 
 ### Integrating a KMS plugin with the remote KMS
 
 The KMS plugin can communicate with the remote KMS using any protocol supported by the KMS.
-All configuration data, including authentication credentials the KMS plugin uses to communicate with the remote KMS, 
+All configuration data, including authentication credentials the KMS plugin uses to communicate with the remote KMS,
 are stored and managed by the KMS plugin independently.
-The KMS plugin can encode the ciphertext with additional metadata that may be required before sending it to the KMS for decryption.
+The KMS plugin can encode the ciphertext with additional metadata that may be required before sending it to the KMS
+for decryption (KMS v2 makes this process easier by providing a dedicated `annotations` field).
```

Comment:
may help to note a usecase/example that uses the annotations field

Reply:
1. You could store the nonce used in the encryption in the annotations field instead of having it be part of the returned ciphertext.
2. If there was some audit ID in the remote KMS to was tied to the particular encryption, it could be stored in an annotation.
3. ...

Reply:
added